aten/src/ATen/cpu/vec/vec256/vec256_bfloat16.h

```diff
@@ -21,6 +21,40 @@ inline namespace CPU_CAPABILITY {
 
 #if defined(CPU_CAPABILITY_AVX2) && !defined(_MSC_VER)
 
+template <typename T, typename std::enable_if_t<is_reduced_floating_point_v<T>, int> = 0>
+inline T down_scale(float val);
+
+template<> inline BFloat16 down_scale(float val) {
+  return BFloat16(val);
+}
+
+template<> inline Half down_scale(float val) {
+  unsigned short d;
+  __m256 v = _mm256_set_ps(val, val, val, val, val, val, val, val);
+  __m128i o = _mm256_cvtps_ph(
+      v, (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC));
+  std::memcpy(&d, &o, sizeof(d));
```

Comment:
How does it compare to a vector store to an array and get the first item of it? Have you checked the assembly code?

Reply:
std::memcpy and a vector store have almost the same performance. Used vector store instead.
I checked the results with this conversion on maxpool https://github.com/pytorch/pytorch/pull/101379 (include avx2 and avx512).

Reply:
good job, you get rid of the memcpy, that's what i was talking about ~